File: Java/Desktop/NewLab8/src/by/bntu/fitr/povt/cng/model/util/logging/AbstractLogger.java
package by.bntu.fitr.povt.cng.model.util.logging;

import by.bntu.fitr.povt.cng.model.util.logging.Decorator.TextDecorator;

import java.util.Date;

public abstract class AbstractLogger implements Logger {

    public AbstractLogger(TextDecorator textDecorator){

    }

    public AbstractLogger() {
    }

    @Override
    public String getBaseString() {
        return new Date().toString() + ": ";
    }

    @Override
    public abstract void logEvent(String string);
}

File: Kotlin/TryKotlin/src/by/bntu/fitr/povt/cng/model/loggers/AbstractLogger.kt
package by.bntu.fitr.povt.cng.model.loggers

import by.bntu.fitr.povt.cng.model.loggers.Logger
import java.util.*

abstract class AbstractLogger : Logger{

    protected fun getDate() : String{
        return Date().toString()
    }

    abstract override fun logEvent(event: String)
}